crates/mpc-protocols/src/mpc_service.rs

```diff
@@ -51,8 +70,30 @@ impl MPCService {
     /// The [`MPCService`] will forward any message related to that instance to this channel.
     fn spawn_mpc_messages_handler(&self, mut receiver: mpsc::Receiver<MPCInput>) {
         tokio::spawn(async move {
+            let language_public_parameters =
+                language_public_parameters::<{ maurer::SOUND_PROOFS_REPETITIONS }>();
+            let party: proof::aggregation::asynchronous::Party<
+                maurer::Proof<{ maurer::SOUND_PROOFS_REPETITIONS }, Lang, PhantomData<()>>,
+            > = proof::aggregation::asynchronous::Party::new_proof_round_party(
+                language_public_parameters.clone(),
+                PhantomData,
+                3,
+                1,
+                &mut OsRng,
+            )
+            .unwrap();
```

Comment:
These public parameters look like they're independent of the instance, and will be the same for every MPC Session. Is it not cheaper to hold only one copy of them in the RAM and only give an `Arc` reference (or whatever type of reference is suitable in the async environment) to them to each instance?

Reply:
We changed the code to initiate the public parameters just once. `Party::new_proof_round_party` function takes ownership so we must clone it.

Reply:
Okay, that saves the computation of the public parameters. But when you're cloning it you're also creating a new copy and therefore using more RAM, no?

(I say this from an efficiency perspective, maybe we don't care about saving that much RAM.)